src/main/java/duke/task/DukeTask.java

```diff
@@ -0,0 +1,77 @@
+package duke.task;
+
+import java.util.Objects;
+
+/**
+ * Represents a generic DukeTask.
+ * It contains a <code>String</code> description describing the task,
+ * a <code>boolean</code> isDone to denote if the task is completed.
+ */
+public abstract class DukeTask {
+    private final String description;
+    private boolean isDone;
+
+    public DukeTask(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns a <code>String</code> representation of isDone.
+     * If the task is completed, it returns a tick denoted by "\u2713".
+     * Else, it returns a cross denoted by "\u2718".
+     *
+     * @return String status icon.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "\u2713" : "\u2718"); //return tick or X symbols
+    }
```

Comment:
Its good that your UI has such a User friendly display! Maybe you can instead define these as private final constant variables above as class level members since these look like Magic Numbers that may confuse future reviewers!